### ISSUE TITLE
Invalid nullability mismatch warning on generic notnull ref parameter discard

### ISSUE THREAD

**Issue by jkoritzinsky** [dotnet/roslyn#82919]
**Version Used**: 5.6.0-2.26161.119

**Steps to Reproduce**:

```csharp
#nullable enable

class Gen<T> where T : notnull
{
    public static void Func(ref T t)
    {
        _ = ref t; // Diagnostic CS8619 here.
    }
}
```

A minimal repro, with source-code provided, is ideal.  Most compiler/language issues can be distilled into a snippet that can be pasted into [.NET Lab](https://lab.razor.fyi/#csharp).

**Diagnostic Id**: CS8619

**Expected Behavior**: No diagnostic

**Actual Behavior**: `CS8619: Nullability of reference types in value of type 'T' doesn't match target type 'T?'`
